### PR TITLE
Feat/vault incentives testtube

### PR DESCRIPTION
**Pull request by magiodev**
## 4. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

**Review by magiodev — COMMENTED** [review]
File by file revision lgtm

**Review by ajansari95 — COMMENTED** [review]
overall looks good two small suggestions

File: smart-contracts/contracts/merkle-incentives/src/admin/execute.rs [hunk 1]
```diff
@@ -20,22 +20,19 @@ pub fn execute_admin_msg(
     deps: DepsMut,
     env: Env,
     info: MessageInfo,
-    admin_msg: AdminExecuteMsg,
+    msg: AdminExecuteMsg,
 ) -> Result<Response, ContractError> {
-    is_contract_admin(&deps.querier, &env, &info.sender)?;
```
[the remark this answers is not in the document]
Comment (magiodev) [reply smart-contracts/contracts/merkle-incentives/src/admin/execute.rs:25]: To update the merkle root we want to allow only the `incentives_admin`. While to update the `incentives_admin` only the `wasm_admin`. I have noted those discussion points anyway, this merge is still a draft, only wants only to merge against the other feature branch for now.

File: smart-contracts/contracts/cl-vault/src/contract.rs [hunk 2]
```diff
@@ -193,7 +184,8 @@ pub fn reply(deps: DepsMut, env: Env, msg: Reply) -> Result<Response, ContractEr
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
-    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
-
+    REWARDS_STATUS.save(deps.storage, &RewardsStatus::Ready)?;
+    DISTRIBUTED_REWARDS.save(deps.storage, &CoinList::new())?;
+    CURRENT_TOTAL_SUPPLY.save(deps.storage, &Uint128::zero())?;
     Ok(Response::new().add_attribute("migrate", "successful"))
 }
```
[the remark this answers is not in the document]
Comment (magiodev) [reply smart-contracts/contracts/cl-vault/src/contract.rs:191]: We still have this on `main` i think. But we will take care of this when willing to merge against `main`.

**Pull request merged**